Rakefile

```diff
@@ -14,12 +16,25 @@ end
 
 namespace :design_system do
   desc "All Design System Tests"
-  task :all do
+  task all: [:report_folders] do
     puts "Running all design system tests"
     system(
       "cd ./testing && bundle exec cucumber -p reports && cd .."
     ) || raise
   end
+
+  task :report_folders do
+    puts "Creating folder structure for this test run"
+    raise "BROWSER must be set" unless ENV["BROWSER"]
+
+    base_path = "artifacts/#{ENV['BROWSER']}/#{ENV.fetch('BROWSERSTACK_CONFIGURATION_OPTIONS', 'grid')}"
+    [
+      "#{base_path}/html_pages",
+      "#{base_path}/logs",
+      "#{base_path}/reports",
+      "#{base_path}/screenshots"
+    ].each { |dir| FileUtils.mkdir_p(dir) }
```

Comment:
I'm not familiar with this. But does this complain if the directories already exist?

Reply:
Nope, it just makes sure they are there.